opencti-platform/opencti-front/src/private/components/common/lists/FilterAutocomplete.tsx

```diff
@@ -0,0 +1,150 @@
+  },
+}));
+
+interface OptionValue extends Option {
+  type: string
+  group?: string
+}
+
+interface FilterAutocompleteProps {
+  filterKey: string
+  defaultHandleAddFilter: HandleAddFilter
+  inputValues: Record<string, string | Date>
+  setInputValues: Dispatch<Record<string, string | Date>>
+  availableEntityTypes?: string[]
+  availableRelationshipTypes?: string[]
+  allEntityTypes?: boolean
+  openOnFocus?: boolean
+}
+
+const FilterAutocomplete: FunctionComponent<FilterAutocompleteProps> = ({
+  filterKey,
+  defaultHandleAddFilter,
+  inputValues,
+  setInputValues,
+  availableEntityTypes,
+  availableRelationshipTypes,
+  allEntityTypes,
+  openOnFocus,
+}) => {
+  const { t } = useFormatter();
+  const classes = useStyles();
+
+  const [searchScope, setSearchScope] = useState<Record<string, string[]>>({});
+
+  const [entities, searchEntities] = useSearchEntities({
+    searchScope,
+    setInputValues,
+    availableEntityTypes,
+    availableRelationshipTypes,
+    allEntityTypes,
+  }) as [Record<string, OptionValue[]>, (filterKey: string, event: SyntheticEvent) => Record<string, OptionValue[]>]; // change when useSearchEntities will be in TS
+
+  const isStixObjectTypes = ['elementId', 'fromId', 'toId', 'objectContains'].includes(filterKey);
+
+  const handleChange = (event: SyntheticEvent, value: AutocompleteValue<OptionValue, boolean | undefined, boolean | undefined, boolean | undefined>) => {
+    if (value && !Array.isArray(value) && typeof value !== 'string') {
+      const group = !onlyGroupOrganization.includes(filterKey)
+        ? value.group
+        : undefined;
+      const filterAdd = `${filterKey}${group ? `_${group}` : ''}`;
+      defaultHandleAddFilter(filterAdd, value.value, value.label, event);
+    }
+  };
+
+  const renderSearchScopeSelection = (key: string) => (
+    <SearchScopeElement
+      name={key}
+      searchScope={searchScope}
+      setSearchScope={setSearchScope}
+    />
+  );
+
+  let options: OptionValue[] = [];
+  if (isStixObjectTypes) {
+    if (searchScope[filterKey] && searchScope[filterKey].length > 0) {
+      options = (entities[filterKey] || [])
+        .filter((n) => (searchScope[filterKey] || []).includes(n.type))
+        .sort((a, b) => (b.type ? -b.type.localeCompare(a.type) : 0));
+    } else {
+      options = (entities[filterKey] || []).sort((a, b) => (b.type ? -b.type.localeCompare(a.type) : 0));
+    }
+  } else if (entities[filterKey]) {
+    options = entities[filterKey];
+  }
+
+  return (
+    <Autocomplete
+      key={filterKey}
+      selectOnFocus={true}
+      openOnFocus={openOnFocus}
+      autoSelect={false}
+      autoHighlight={true}
+      getOptionLabel={(option) => (option.label ?? '')}
+      noOptionsText={t('No available options')}
+      options={options}
+      onInputChange={(event) => searchEntities(filterKey, event)}
+      inputValue={inputValues[filterKey] as string || ''}
+      onChange={(event, value) => handleChange(event, value)}
```

Comment:
Need to type directly the onChange (event: SyntheticEvent, value: Option | null) => 
to have a clear handleChange type.